### PR TITLE
CMake: update porting guide after target refactor

### PR DESCRIPTION
**Pull request by 0xc0170**
The recent Mbed OS release updated the CMake targets but not the docs. This PR should address that.

cc @hugueskamba @rajkan01

### CODE REVIEW

File: docs/porting/target/cmake.md [hunk 1]
```diff
@@ -15,53 +15,84 @@ Note: All the CMake variables above are generated by mbed-tools and their values
 
 ## Mbed targets CMake input file structure
 
+We still support mbed-cli 1 and its requirements for the folders naming. To stay backward compatible, we still use the same naming even with CMake.
```
Comment (hugueskamba) [inline docs/porting/target/cmake.md]:
```suggestion
As we are still supporting Mbed CLI 1, we have maintained the directory naming scheme that use prefixes (`FEATURE_`, `COMPONENT_`, `TARGET_`).
```

File: docs/porting/target/cmake.md [hunk 2]
```diff
@@ -15,53 +15,84 @@ Note: All the CMake variables above are generated by mbed-tools and their values
...
+
+Each Mbed OS target should be prefixed with `mbed-`. We translate automatically Mbed OS target names from targets.json to CMake targets. A target from targets.json named `CYSBSYSKIT_01` has CMake target `mbed-cysbsyskit-01`.
```
Comment (hugueskamba) [inline docs/porting/target/cmake.md]:
```suggestion
Every Mbed CMake target should be prefixed with `mbed-`. Mbed boards listed in `targets.json` are translated in CMake to add the prefix and replace `_` with `-`. For example, the Mbed board `CYSBSYSKIT_01` is represented by the CMake target `mbed-cysbsyskit-01`.
```

File: docs/porting/target/cmake.md [hunk 3]
```diff
@@ -15,53 +15,84 @@ Note: All the CMake variables above are generated by mbed-tools and their values
...
+
 ### Vendor selection
 
-Vendor selection is handled in [`targets/CMakeLists.txt`](https://github.com/ARMmbed/mbed-os/blob/master/targets/CMakeLists.txt) using a conditional statement to verify that one of the labels used by a given Mbed target matches a given partner directory. The matching directory is added to directories to build.
+Vendor selection is handled in [`targets/CMakeLists.txt`](https://github.com/ARMmbed/mbed-os/blob/master/targets/CMakeLists.txt). The matching directory is added to directories to build.
```
Comment (hugueskamba) [inline docs/porting/target/cmake.md]:
```suggestion
Vendor selection is handled in [`targets/CMakeLists.txt`](../../targets/CMakeLists.txt). A subdirectory is added to the build if one of the CMake targets it contains is referenced.
```

File: docs/porting/target/cmake.md [hunk 4]
```diff
@@ -15,53 +15,84 @@ Note: All the CMake variables above are generated by mbed-tools and their values
...
+add_subdirectory(common/COMPONENT_WHD EXCLUDE_FROM_ALL)
+
+add_subdirectory(TARGET_CY8CKIT064B0S2_4343W EXCLUDE_FROM_ALL)
+add_subdirectory(TARGET_CY8CKIT_062S2_43012 EXCLUDE_FROM_ALL)
 ...
-endif()
+
+add_library(mbed-psoc6 INTERFACE)
 
 # Add the include directories accessible from this directory that are not specific to an MCU family or Mbed target
```
Comment (hugueskamba) [inline docs/porting/target/cmake.md]:
```suggestion
# Add the include directories accessible from this directory that are specific to this MCU family or Mbed target
```

File: docs/porting/target/cmake.md [hunk 5]
```diff
@@ -15,53 +15,84 @@ Note: All the CMake variables above are generated by mbed-tools and their values
...
-target_include_directories(mbed-core
+target_include_directories(mbed-psoc6
     INTERFACE
         ...
 )
 
 # Add the source files accessible from this directory that are not specific to an MCU family or Mbed target
```
Comment (hugueskamba) [inline docs/porting/target/cmake.md]:
```suggestion
# Add the source files accessible from this directory that are specific to this MCU family or Mbed target
```

File: docs/porting/target/cmake.md [hunk 6]
```diff
@@ -15,53 +15,84 @@ Note: All the CMake variables above are generated by mbed-tools and their values
...
-target_sources(mbed-core
+target_sources(mbed-psoc6
     INTERFACE
         ...
 )
 ```
 
+The board target can link to `mbed-psoc6`. Let's take `mbed-cysbsyskit-01` target:
```
Comment (hugueskamba) [inline docs/porting/target/cmake.md]:
```suggestion
The board CMake targets should link to `mbed-psoc6`:
```

File: docs/porting/target/cmake.md [hunk 7]
```diff
@@ -15,53 +15,84 @@ Note: All the CMake variables above are generated by mbed-tools and their values
...
+
+```
+add_library(mbed-cysbsyskit-01 INTERFACE)
+
+target_include_directories(mbed-cysbsyskit-01
+    INTERFACE
+        .
+        device
```
Comment (hugueskamba) [inline docs/porting/target/cmake.md]:
```suggestion
        ...
```

File: docs/porting/target/cmake.md [hunk 8]
```diff
@@ -15,53 +15,84 @@ Note: All the CMake variables above are generated by mbed-tools and their values
...
+)
+
+target_sources(mbed-cysbsyskit-01
+    INTERFACE
+        PeripheralPins.c
+        cybsp.c
```
Comment (hugueskamba) [inline docs/porting/target/cmake.md]:
```suggestion
        ...
```

File: docs/porting/target/cmake.md [hunk 9]
```diff
@@ -15,53 +15,84 @@ Note: All the CMake variables above are generated by mbed-tools and their values
...
+)
+
```
Comment (hugueskamba) [inline docs/porting/target/cmake.md]:
```suggestion
# Other libraries that the Mbed board depends on
```

File: docs/porting/target/cmake.md [hunk 10]
```diff
@@ -15,53 +15,84 @@ Note: All the CMake variables above are generated by mbed-tools and their values
...
+target_compile_definitions(mbed-cysbsyskit-01
+    INTERFACE
+        "CY8C624AFNI_S2D43F"
+)
+
+mbed_post_build_psoc6_merge_hex("CYSBSYSKIT_01")
+```
+
 ## Add target's CMakeLists.txt
```
Comment (hugueskamba) [inline docs/porting/target/cmake.md]:
```suggestion
## Add Mbed board CMake input source file
```

File: docs/porting/target/cmake.md (hunk 3 above)
Comment (0xc0170) [reply docs/porting/target/cmake.md]: `../../targets/CMakeLists.txt` is this valid url (does this work within docs) ?

Comment (hugueskamba) [reply docs/porting/target/cmake.md:24]: Try you will see. It's a relative link.

Comment (hugueskamba) [reply docs/porting/target/cmake.md]: No it will not. I thought this file was in Mbed OS.

Comment (0xc0170) [reply docs/porting/target/cmake.md]: I'll keep it as it is.

This is documentation (one of the things we will consolidate one day to keep code and docs in one place).

Comment (hugueskamba) [reply docs/porting/target/cmake.md]: There is a suggestion for the second sentence `A subdirectory is added to the build if one of the CMake targets it contains is referenced.`

Comment (0xc0170) [reply docs/porting/target/cmake.md]: Fixing now

**Review by hugueskamba — APPROVED** [review]

**Comment by 0xc0170** [comment]
@donatieng Can we get this in for the upcoming release?

**Pull request merged**